src/platform/plugins/shared/share/server/unused_urls_task/task.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Duration } from 'moment';
+import { CoreSetup, ISavedObjectsRepository, SavedObjectsBulkDeleteObject } from '@kbn/core/server';
+import { Logger } from '@kbn/logging';
+import { TaskInstanceWithId } from '@kbn/task-manager-plugin/server/task';
+import { TaskManagerStartContract } from '@kbn/task-manager-plugin/server';
+import { SavedObjectsUtils } from '@kbn/core-saved-objects-utils-server';
+import { SAVED_OBJECT_TYPE, TASK_ID } from './constants';
+
+export const durationToSeconds = (duration: Duration) => `${duration.asSeconds()}s`;
+
+export const getDeleteUnusedUrlTaskInstance = (interval: Duration): TaskInstanceWithId => ({
+  id: TASK_ID,
+  taskType: TASK_ID,
+  params: {},
+  state: {},
+  schedule: {
+    interval: durationToSeconds(interval),
+  },
+  scope: ['share'],
+});
+
+export const deleteUnusedUrls = async ({
+  savedObjectsRepository,
+  unusedUrls,
+  namespace,
+  logger,
+}: {
+  savedObjectsRepository: ISavedObjectsRepository;
+  unusedUrls: SavedObjectsBulkDeleteObject[];
+  namespace: string;
+  logger: Logger;
+}) => {
+  try {
+    logger.debug(`Deleting ${unusedUrls.length} unused URL(s) in namespace "${namespace}"`);
+
+    await savedObjectsRepository.bulkDelete(unusedUrls, {
```

Comment:
we are gonna attempt bulk deletion, ignore the result and proceed to the next page -- I think this makes sense given the "best effort" nature of this cleanup task 👍🏻